plugins/catalog-backend/src/database/Database.test.ts

```diff
@@ -14,30 +14,74 @@
  * limitations under the License.
  */
 
-import knex from 'knex';
+import {
+  ConflictError,
+  getVoidLogger,
+  NotFoundError,
+} from '@backstage/backend-common';
+import Knex from 'knex';
 import path from 'path';
 import { Database } from './Database';
-import { AddDatabaseLocation, DbLocationsRow } from './types';
+import {
+  AddDatabaseLocation,
+  DbEntityRequest,
+  DbEntityResponse,
+  DbLocationsRow,
+} from './types';
 
 describe('Database', () => {
-  const database = knex({
-    client: 'sqlite3',
-    connection: ':memory:',
-    useNullAsDefault: true,
-  });
-  database.client.pool.on('createSuccess', (_eventId: any, resource: any) => {
-    resource.run('PRAGMA foreign_keys = ON', () => {});
-  });
+  let database: Knex;
+  let entityRequest: DbEntityRequest;
+  let entityResponse: DbEntityResponse;
 
   beforeEach(async () => {
+    database = Knex({
+      client: 'sqlite3',
+      connection: ':memory:',
+      useNullAsDefault: true,
+    });
+
+    await database.raw('PRAGMA foreign_keys = ON');
     await database.migrate.latest({
       directory: path.resolve(__dirname, 'migrations'),
       loadExtensions: ['.ts'],
     });
+
+    entityRequest = {
+      entity: {
+        apiVersion: 'a',
+        kind: 'b',
+        metadata: {
+          name: 'c',
+          namespace: 'd',
+          labels: { e: 'f' },
+          annotations: { g: 'h' },
+        },
+        spec: { i: 'j' },
+      },
+    };
+
+    entityResponse = {
+      locationId: undefined,
+      entity: {
+        apiVersion: 'a',
+        kind: 'b',
+        metadata: {
+          uid: expect.anything(),
+          etag: expect.anything(),
+          generation: expect.anything(),
+          name: 'c',
+          namespace: 'd',
+          labels: { e: 'f' },
+          annotations: { g: 'h' },
+        },
+        spec: { i: 'j' },
+      },
+    };
   });
 
   it('manages locations', async () => {
-    const db = new Database(database);
+    const db = new Database(database, getVoidLogger());
```

Comment:
Is it worth making the `Database` default to using a `VoidLogger`?

Reply:
Maybe - it could be `constructor(database: KnexWhatever, options?: DatabaseOptions)` or so i guess? But perhaps as a later iteration in that case?